Content.Server/Power/EntitySystems/ExtensionCableSystem.cs

```diff
@@ -258,14 +260,18 @@ private bool TryFindAvailableProvider(EntityUid owner, float range, [NotNullWhen
 
                 if (!provider.Connectable) continue;
 
-                if ((Transform(entity).LocalPosition - xform.LocalPosition).Length > Math.Min(range, provider.TransferRange)) continue;
+                var distance = (Transform(entity).LocalPosition - xform.LocalPosition).Length;
+                if (distance > Math.Min(range, provider.TransferRange)) continue;
 
-                foundProvider = provider;
-                return true;
+                candidates.Add(new Tuple<float, ExtensionCableProviderComponent>(distance, provider));
             }
 
-            foundProvider = default;
-            return false;
+            foundProvider = candidates
```

Comment:
Since you already have a `List<T>` you should use `Sort()` to sort it in-place, much faster.